### PR TITLE
Added new Transfer node

### LINKED ISSUE #449: Compound node for transfers
I've been using a compound node which contains a storage with zero volume to reduce the number of routes in highly interconnected models.

The node looks like this, where X and Y are actually many nodes:

```
X ->- | -- storage ->- link -- | ->- Y
```

Internally the storage is an input and output which crucially are *not* connected. This means that for X1..3 and Y1..3 there are 6 routes in the model (3+3), not 9 (3*3).

The node uses iter_slots to expose the correct node when connecting, so you can do this:

```
T = Transfer(max_flow=1.5, cost=10)
X1.connect(T)
T.connect(Y1)
```

I'll submit a PR for this on Monday. What do you think about the node name `Transfer`? It's really applicable anywhere you want to collapse the routes, not just inter-zone transfers (which is what it was designed for).

### PR DESCRIPTION
**Pull request by snorfalorpagus**
This PR adds a compound transfer node, composed of a Storage with zero volume and a Link.

Closes #449.

### CODE REVIEW

File: pywr/nodes.py [hunk 1]
```diff
@@ -746,4 +746,86 @@ def __init__(self, model, name, nodes, **kwargs):
         super(AggregatedNode, self).__init__(model, name, **kwargs)
         self.nodes = nodes
 
+class Transfer(Node):
```
Comment (jetuk) [inline pywr/nodes.py]: I think this is a good implementation. My only comment is whether to stick with the name `Transfer`. This is a quite generic extenstion to the basic `Link`. Should it derive its name from that instead?

Comment (snorfalorpagus) [inline pywr/nodes.py]: If we can think of something better, yes. Transfer doesn't really describe what it does either, just what we've used it for.

ZeroStorageLink?
NullStorage?
BreakLink?

Comment (jetuk) [inline pywr/nodes.py]: BufferedLink?

Comment (snorfalorpagus) [inline pywr/nodes.py]: Buffered implies some carryover between timesteps to me.

Comment (jetuk) [inline pywr/nodes.py]: True. I guess BreakLink then?

Comment (snorfalorpagus) [inline pywr/nodes.py]: We agreed on `BreakLink`.

**Pull request merged**